pvlib/losses.py

```diff
@@ -9,13 +9,20 @@
 import pandas as pd
 from pvlib.tools import cosd
 
+_RAIN_ACC_PERIOD = pd.Timedelta('1h')
```

Comment:
What is the argument for these new constants? They aren't reused and less explicit at the API than e,g.  `pd.Timedelta('1h'). 

Nit: _RAIN_ACCUM_PERIOD to keep the constant the same as the kwarg

Reply:
For the deposition velocity, I believe having a mutable default input arg is a Python no-no, I've seen linters ding this, and there are blog posts that also say to avoid this pattern.

For the default rain accumulation period, I'm not sure, and I'm probably wrong, but I feel like having a function/class as a input arg default is an extra burden. Why not just call it once? Although, maybe behind the scenes that's exactly what Python does when it imports a module? I don't know. I agree with you on readability tho, so I'm okay with reverting this, it's not a blocker for me, just a preference. Also, "premature optimisation ..." Sorry

Reply:
The dict as default was probably my idea, out of ignorance. Are you willing to separate `depo_veloc` into two separate inputs? That would be more explicit and avoids hiding the default in a private constant.

Reply:
> maybe behind the scenes that's exactly what Python does 

Yep, default arguments are evaluated and stored when the function is defined, meaning default arguments are created once and re-used forever after.  The argument against mutable defaults is that changes in one function invocation persist to the next: 

```
In [18]: def foo(lst=[], ts=datetime.datetime.now()):
    ...:     lst.append(5)
    ...:     print(ts, lst)
    ...: 
    ...: 
    ...: foo()
    ...: time.sleep(1)
    ...: foo()
    ...: 
2020-02-24 10:52:10.714337 [5]
2020-02-24 10:52:10.714337 [5, 5]
```

So strictly speaking it's okay to have mutable defaults as long as you never mutate them.  But avoiding them at all will ensure you never accidentally mutate them by accident.

Reply:
I used to be a stickler about no mutable default arguments but I now think they're ok if used responsibly. My preferences are in order:

1. split dict into two different kwargs
2. leave the dict alone
3. set kwarg to `None`, set default value within function
4. set kwarg to `None`, use default value from private constant

I'm -1 on the change to the `rain_accum_period` kwarg.

Reply:
Thanks Kevin for the explanation!

>Are you willing to separate `depo_veloc` into two separate inputs? That would be more explicit and avoids hiding the default in a private constant.

The pattern I've seen implemented most often is to put `None` as the default or some other appropriate constant, then check for it and set the default inside the scope of the function. IMO having a lot of arguments makes the function harder to use and there's a linter warning for that too.

I don't think we should depend on the function signature to inform the user about the API, that should be the docstring's job, although ideally they are consistent, and arguably having explicit defaults is more readable. So these are just my opinions, I don't really have a strong feeling about what's best. Paint the bike shed white ... Sorry

Reply:
> put None as the default or some other appropriate constant, then check for it and set the default inside the scope of the function. 

This. Was just reading a blog post which recommended this approach.